pixeldata/src/lib.rs

```diff
@@ -865,8 +737,172 @@ mod tests {
+                .unwrap()
+                .to_ndarray::<u16>()
+                .unwrap();
+            // Validated using Numpy
+            // This doesn't reshape the array based on the PlanarConfiguration
+            // So for this scan the pixel layout is [Rlsb..Rmsb, Glsb..Gmsb, Blsb..msb]
+            assert_eq!(ndarray.shape(), &[2, 100, 100, 3]);
+            assert_eq!(ndarray.len(), 60000);
+            // The second frame is the inverse of the first frame
+            assert_eq!(ndarray[[1, 0, 0, 0]], 0);
+            assert_eq!(ndarray[[1, 0, 0, 1]], 0);
+            assert_eq!(ndarray[[1, 0, 0, 2]], 0);
+            assert_eq!(ndarray[[1, 50, 50, 0]], 32639);
+            assert_eq!(ndarray[[1, 50, 50, 1]], 32639);
+            assert_eq!(ndarray[[1, 50, 50, 2]], 32639);
+            assert_eq!(ndarray[[1, 75, 75, 0]], 65535);
+            assert_eq!(ndarray[[1, 75, 75, 1]], 65535);
+            assert_eq!(ndarray[[1, 75, 75, 2]], 65535);
+        }
+
+        #[rstest]
+        // jpeg2000 encoding not supported
+        #[should_panic(expected = "UnsupportedTransferSyntax { ts: \"1.2.840.10008.1.2.4.91\"")]
+        #[case("pydicom/693_J2KI.dcm", 1)]
+        #[should_panic(expected = "UnsupportedTransferSyntax { ts: \"1.2.840.10008.1.2.4.90\"")]
+        #[case("pydicom/693_J2KR.dcm", 1)]
+        //
+        // jpeg-ls encoding not supported
+        #[should_panic(expected = "UnsupportedTransferSyntax { ts: \"1.2.840.10008.1.2.4.80\"")]
+        #[case("pydicom/emri_small_jpeg_ls_lossless.dcm", 10)]
+        #[should_panic(expected = "UnsupportedTransferSyntax { ts: \"1.2.840.10008.1.2.4.80\"")]
+        #[case("pydicom/MR_small_jpeg_ls_lossless.dcm", 1)]
+        //
+        // sample precicion of 12 not supported
+        #[should_panic(expected = "Unsupported(SamplePrecision(12))")]
+        #[case("pydicom/JPEG-lossy.dcm", 1)]
+        //
+        // works, but output has coloroffset to what is produced by other tools
+        #[case("pydicom/color3d_jpeg_baseline.dcm", 120)]
+        //
+        // Works fine
+        #[case("pydicom/JPEG-LL.dcm", 1)]
+        #[case("pydicom/JPGLosslessP14SV1_1s_1f_8b.dcm", 1)]
+        #[case("pydicom/SC_rgb_jpeg_gdcm.dcm", 1)]
+        #[case("pydicom/SC_rgb_jpeg_lossy_gdcm.dcm", 1)]
+
+        fn test_parse_jpeg_encoded_dicom_pixel_data(#[case] value: &str, #[case] frames: u16) {
+            let test_file = dicom_test_files::path(value).unwrap();
+            println!("Parsing pixel data for {}", test_file.display());
+            let obj = open_file(test_file).unwrap();
+            let pixel_data = obj.decode_pixel_data().unwrap();
+            assert_eq!(pixel_data.number_of_frames, frames);
+
+            let output_dir = Path::new(
+                "../target/dicom_test_files/_out/test_parse_jpeg_encoded_dicom_pixel_data",
+            );
+            fs::create_dir_all(output_dir).unwrap();
+
+            for i in 0..pixel_data.number_of_frames {
```

Comment:
Fetch less frames to make the test faster.
```suggestion
            for i in 0..pixel_data.number_of_frames.min(16) {
```

Reply:
I assume you mean .take here?